internal/test/e2e/run.go

```diff
@@ -482,3 +471,35 @@ func logTestGroups(logger logr.Logger, instancesConf []instanceRunConf) {
 	}
 	logger.V(1).Info("Running tests in parallel", "testsGroups", testGroups)
 }
+
+func getHardwarePool(storageBucket string) ([]*api.Hardware, error) {
+	awsSession, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("creating aws session for tests: %v", err)
+	}
+	err = s3.DownloadToDisk(awsSession, os.Getenv(tinkerbellHardwareS3FileKeyEnvVar), storageBucket, e2eHardwareCsvFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download tinkerbell hardware csv: %v", err)
+	}
+
+	hardware, err := api.NewHardwareSliceFromFile(e2eHardwareCsvFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Tinkerbell hardware: %v", err)
+	}
+	return hardware, nil
+}
+
+func reserveTinkerbellHardware(conf instanceRunConf, invQueue *hardwareQueue) error {
+	reservedTinkerbellHardware, err := invQueue.reserveHardware(conf.hardwareCount)
+	if err != nil {
+		return fmt.Errorf("TestRunner timed out waiting for hardware - test instanceId: %v", conf.instanceId)
+	}
+	conf.hardware = reservedTinkerbellHardware
+	conf.logger.V(1).Info("Reserved hardware for TestRunner", "instanceId", conf.instanceId, "hardwarePool", conf.hardware)
```

Comment:
Lets not print the whole conf.hardware list. Lets just print the names of the hw?